### PR TITLE
feat: Testnet Rentals function to return assets to their respective lessors

### LINKED ISSUE #78: Add a function only for goerli that allows the owner to return assets to their respective lessors

### PR DESCRIPTION
**Pull request by fzavalia**
Closes #78 

Added a function in ExtendedRentals that allows the owner to return all assets to their respective lessors.

This required the rentals mapping to be changed from `private` to `internal`

ExtendedRentals will be used only in testnets and that function is to facilitate development in the case the contract has to be redeployed

### CODE REVIEW

File: contracts/mocks/ExtendedRentals.sol [hunk 1]
```diff
@@ -4,13 +4,54 @@ pragma solidity ^0.8.7;
 
 import "../Rentals.sol";
 
-/// @dev Extension of the Rentals contract to expose some internal elements for testing.
+/// @notice Extension of the Rentals contract.
+/// @dev This contract is used for testing purposes only and should not be used in production.
 contract ExtendedRentals is Rentals {
+    // Event emitted when a rented asset is returned to its lessor via "returnToLessor".
+    event AssetReturned(address indexed _contractAddress, uint256 indexed _tokenId, address _lessor, address _sender);
```
Comment (nachomazzara) [inline contracts/mocks/ExtendedRentals.sol:11]: Why we need this?

File: test/Rentals.spec.ts [hunk 2]
```diff
@@ -3908,4 +3908,70 @@ describe('Rentals', () => {
       await expect(rentals.connect(owner).unpause()).to.be.revertedWith('Pausable: not paused')
     })
   })
+
+  describe('returnToLessor :: ExtendedRentals', () => {
```
Comment (nachomazzara) [inline test/Rentals.spec.ts:3912]: IMO  It is not a good practise to test mock contracts that are not a copy of what we will have in production

File: contracts/mocks/ExtendedRentals.sol (hunk 1 above)
Comment (fzavalia) [reply contracts/mocks/ExtendedRentals.sol:11]: Not really needed, just informational

File: test/Rentals.spec.ts (hunk 2 above)
Comment (fzavalia) [reply test/Rentals.spec.ts:3912]: Do you suggest the removal of the test or moving it elsewhere?

File: contracts/mocks/ExtendedRentals.sol (hunk 1 above)
Comment (fzavalia) [reply contracts/mocks/ExtendedRentals.sol]: Removed!

File: test/Rentals.spec.ts (hunk 2 above)
Comment (fzavalia) [reply test/Rentals.spec.ts:3912]: I'll leave the test here. 
We could later discuss a way in which we could standardize/handle cases like this were testnet might have additional "helper" functions

Comment (nachomazzara) [reply test/Rentals.spec.ts:3912]: Ok, then we should follow the good practice and remove the tests. Here is not the place to keep test mocked smart contracts

Comment (fzavalia) [reply test/Rentals.spec.ts]: Sure

**Comment by fzavalia** [comment]
@nachomazzara Separated ExtendedRentals into DevRentals
Added DevRentals.spec.ts to test the new function apart from the original Rental tests

**Review by nachomazzara — APPROVED** [review]

**Pull request merged**